### ISSUE TITLE
how to normalize data

### ISSUE THREAD

**Issue by roddtalebi** [ezCGP/ezCGP#131]
We had to remove the normalize() primitive because Augmentor.Pipeline uses PIL.Image which requires uint8 format so float values btwn. (0,1) won't work.

Looks like we can use several of the flags in ImageDataGenerator but which ones should we set? I'm terribly not familiar with the recommended strategies
https://www.tensorflow.org/api_docs/python/tf/keras/preprocessing/image/ImageDataGenerator

**Comment by roddtalebi** [comment]
some of the arguments:

featurewise_center | Boolean. Set input mean to 0 over the dataset, feature-wise.
samplewise_center | Boolean. Set each sample mean to 0.
featurewise_std_normalization | Boolean. Divide inputs by std of the dataset, feature-wise.
samplewise_std_normalization | Boolean. Divide each input by its std.

BUT!
here is what it says for preprocessing_function which we use to pass in the Augmentor.Pipeline...

> function that will be applied on each input. The function will run after the image is resized and augmented. The function should take one argument: one image (Numpy tensor with rank 3), and should output a Numpy tensor with the same shape.

Does this mean that we will be unable to use the normalizing methods of ImageDataGen because it will happen before it gets passed into Augmentor.Pipeline?

**Comment by roddtalebi** [comment]
should make a dummy py script to see what ImageDataGen does with the images with each flag. should note that augmentor.pipeline.keras_preprocess_func assumes the data will be coming in as (0,1) floats, converts it to PIL.Image np.uint8, passes them through the pipeline operators BUT does not renormalize or convert back to np

**Comment by roddtalebi** [comment]
Maybe use Norm_Min_Max here?
https://docs.opencv.org/2.4/modules/core/doc/operations_on_arrays.html#cv2.normalize

add it as the last operator before we create the generator so it always gets performed last?
